samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -153,31 +166,46 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
-   * Create the Coordination needed by the application runner.
-   * @return {@link CoordinationUtils}
-   * @throws Exception exception
+   * Block until the application finishes
+   */
+  public void waitForFinish() {
+    try {
+      shutdownLatch.await();
+    } catch (Exception e) {
+      log.error("Wait is interrupted by exception", e);
+      throw new SamzaException(e);
+    }
+  }
+
+  /**
+   * Create the {@link CoordinationUtils} needed by the application runner.
+   * @return an instance of {@link CoordinationUtils}
    */
-  /* package private */ CoordinationUtils getCoordinationUtils() throws Exception {
-    ApplicationConfig appConfig = new ApplicationConfig(config);
-    String clazz = appConfig.getCoordinationServiceFactoryClass();
-    if (clazz != null) {
-      CoordinationServiceFactory factory = ClassLoaderHelper.fromClassName(clazz);
-      return factory.getCoordinationService(appConfig.getGlobalAppId(), uid, config);
+  /* package private */ CoordinationUtils createCoordinationUtils() {
+    String jobCoordinatorFactoryClassName = config.get(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY, "");
```

Comment:
I remembered I got that config exception once, couldn't remember what's the case anymore. Seems it should always be there.